Documentation/gettingstarted/egress-gateway.rst

```diff
@@ -7,159 +7,388 @@
-specific, gateway nodes. Packets are masqueraded to the gateway node IP.
+When the egress gateway feature is enabled and egress gateway policies are in
+place, matching packets that leave the cluster are masqueraded with selected,
+predictable IPs associated with the gateway nodes. As an example, this feature
+can be used in combination with legacy firewalls to allow traffic to legacy
+infrastructure only from specific pods within a given namespace. The pods
+typically have ever-changing IP addresses, and even if masquerading was to be
+used as a way to mitigate this, the IP addresses of nodes can also change
+frequently over time.
 
-This document explains how to enable the egress gateway and configure
-egress NAT policies to route and SNAT the egress traffic for a specific
-workload.
+This document explains how to enable the egress gateway feature and how to
+configure egress gateway policies to route and SNAT the egress traffic for a
+specific workload.
 
 .. note::
 
-   This guide assumes that Cilium has been correctly installed in your
-   Kubernetes cluster. Please see :ref:`k8s_quick_install` for more
-   information. If unsure, run ``cilium status`` and validate that Cilium is up
-   and running.
+    This guide assumes that Cilium has been correctly installed in your
+    Kubernetes cluster. Please see :ref:`k8s_quick_install` for more
+    information. If unsure, run ``cilium status`` and validate that Cilium is up
+    and running.
+
+Preliminary Considerations
+==========================
+
+Cilium must make use of network-facing interfaces and IP addresses present on
+the designated gateway nodes. These interfaces and IP addresses must be
+provisioned and configured by the operator based on their networking
+environment. The process is highly-dependent on said networking environment. For
+example, in AWS/EKS, and depending on the requirements, this may mean creating
+one or more Elastic Network Interfaces with one or more IP addresses and
+attaching them to instances that serve as gateway nodes so that AWS can
+adequately route traffic flowing from and to the instances. Other cloud
+providers have similar networking requirements and constructs.
+
+Additionally, the enablement of the egress gateway feature requires that both
+BPF masquerading and the kube-proxy replacement are enabled, which may not be
+possible in all environments (due to, e.g., incompatible kernel versions).
+
+Compatibility with other features
+=================================
+
+L7 policies
+-----------
+
+Egress gateway is currently partially incompatible with L7 policies .
+Specifically, when an egress gateway policy and a L7 policy both select the same
+endpoint, traffic from said endpoint will not go through egress gateway, even if
+the policy allows it. Full support will be added in an upcoming release once
+:gh-issue:`19642` is resolved.
+
+LB acceleration
+---------------
+
+When egress gateway is used in combination with XDP-based LB acceleration
```

Comment:
```suggestion
When the egress gateway is used in combination with XDP-based LB acceleration
```

Reply:
or "an"?

Reply:
I'll go for "When the egress gateway feature", as we are not referring to any particular not node here